### ISSUE TITLE
Improve pagination for columns

### ISSUE THREAD

**Issue by kristjanstrojan** [Kozea/WeasyPrint#365]
Hi,

I am generating PDF from HTML string and when I declare in css:

`body {columns: 2}`

I get empty PDF.

Am I missing something here?

Best,
Kristjan

**Comment by liZe** [comment]
Thanks for reporting!

Could you please give an example of HTML that doesn't work?

**Comment by kristjanstrojan** [comment]
So,

here is my html:

```html 
<html>

<body>
    <div class="element">
        <div class="a_image" style="background: url('file://tmp/32864.jpg'); background-clip:content-box; background-size:cover;"></div>
    </div>
    <div class="element">
        <div class="a_image" style="background: url('file://tmp/32810.jpg'); background-clip:content-box; background-size:cover;"></div>
    </div>
    <div class="element">
        <div class="a_image" style="background: url('file://tmp/32864.jpg'); background-clip:content-box; background-size:cover;"></div>
    </div>
    <div class="element">
        <div class="a_image" style="background: url('file://tmp/32810.jpg'); background-clip:content-box; background-size:cover;"></div>
    </div>
</body>

</html>
```

and here is the css: 

```css
body {
    column-count: 2
 }
 
.element, .a_image {
    height:4in;
 }
```

This is just on one page and it works OK. But as soon as I add additional  .element element, which should be shown on the next page, it produces empty PDF with no error thrown.

Hope it helps.

**Comment by liZe** [comment]
I get the idea. I can't reproduce your problem exactly (I always get a PDF), but it's easy to get bugs with pagination. It's written in the documentation (ok, ok, that's no excuse).

Simple examples:
```html
<style>
  body { column-count: 2 }
  div { height: 4in }
</style>
<div style="background: blue  ">A</div>
<div style="background: green ">B</div>
<div style="background: orange">C</div>
<div style="background: red   ">D</div>
<div style="background: yellow">E</div>
```

```html
<style>
  body { column-count: 2 }
  div { height: 4in }
</style>
<div style="background: blue  "></div>
<div style="background: green "></div>
<div style="background: orange"></div>
<div style="background: red   "></div>
<div style="background: yellow"></div>
```

These two different bugs are probably quite easy to fix (and to test). Anyone interested?

And for the record: it's definitely related to #60.

**Comment by liZe** [comment]
> These two different bugs are probably quite easy to fix (and to test).

Actually: no.

It would be easy with atomic boxes (with `img` instead of `div` for example), but hard with non-atomic boxes. Cutting boxes correctly is not even supported between pages (see [this test](http://test.weasyprint.org/suite-css21/chapter13/section2/test7/) for example). And with a fixed height it's just crazy, I'm not even sure that the W3C was sadistic enough to include a test like this in their suite.

**Comment by liZe** [comment]
This issue, with #367 too, is fixed by #878.
